crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -114,6 +111,9 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
                         .start_timer();
 
                     let mut watermark_needs_update = false;
+                    // Concurrent pipelines are more lax about succeeding every watermark update, so
```

Comment:
In a couple of week's time we are going to be asking "more lax than what?". Let's phrase this in terms of properties of this pipeline, rather than by comparison with the sequential pipeline:

> Concurrent pipelines may not always produce a watermark update.

...or something.